app/controllers/annotation_categories_controller.rb

```diff
@@ -20,11 +20,13 @@ def index
         data = @annotation_categories.map do |cat|
           {
             id: cat.id,
-            annotation_category_name: cat.annotation_category_name,
+            annotation_category_name: cat.annotation_category_name +
+              (cat.flexible_criterion_id.nil? ? '' : ' [' + cat.flexible_criterion.name + ']' ),
             texts: cat.annotation_texts.map do |text|
               {
                 id: text.id,
-                content: text.content
+                content: text.content,
+                deduction: text.deduction.nil? ? 'undefined' : text.deduction
```

Comment:
Use `nil` here instead of `'undefined'`; this means you can simplify this ternary expression to just `text.deduction`. This occurs a few more times below.